src/entropies_estimators/nearest_neighbors/Lord.jl

```diff
@@ -0,0 +1,167 @@
+using StaticArrays: @MVector, @MMatrix, SVector, MVector
+using Neighborhood: NeighborNumber, KDTree, Theiler, Euclidean
+using Neighborhood: bulksearch
+using LinearAlgebra: svd
+
+export Lord
+
+"""
+    Lord <: DifferentialEntropyEstimator
+    Lord(; k = 10, w = 0)
+
+`Lord` estimates the [`Shannon`](@ref) differential [`entropy`](@ref) using a nearest
+neighbor approach with a local nonuniformity correction (LNC).
+
+`w` is the Theiler window, which determines if temporal neighbors are excluded
+during neighbor searches (defaults to `0`, meaning that only the point itself is excluded
+when searching for neighbours).
+
+## Description
+
+Assume we have samples ``\\bar{X} = \\{\\bf{x}_1, \\bf{x}_2, \\ldots, \\bf{x}_N \\}`` from a
+continuous random variable ``X \\in \\mathbb{R}^d`` with support ``\\mathcal{X}`` and
+density function ``f : \\mathbb{R}^d \\to \\mathbb{R}``. `Lord` estimates the
+[Shannon](@ref) differential entropy
+
+```math
+H(X) = \\int_{\\mathcal{X}} f(x) \\log f(x) dx = \\mathbb{E}[-\\log(f(X))],
+```
+
+by using the resubstitution formula
+
+```math
+\\hat{\\bar{X}, k} = -\\mathbb{E}[\\log(f(X))]
+\\approx \\sum_{i = 1}^N \\log(\\hat{f}(\\bf{x}_i)),
+```
+
+where ``\\hat{f}(\\bf{x}_i)`` is an estimate of the density at ``\\bf{x}_i`` constructed
+in a manner such that ``\\hat{f}(\\bf{x}_i) \\propto \\dfrac{k(x_i) / N}{V_i}``,
+where ``k(x_i)`` is the number of points in the neighborhood of ``\\bf{x}_i``, and ``V_i``
+is the volume of that neighborhood.
+
+While most nearest-neighbor based differential entropy estimators uses regular volume
+elements (e.g. hypercubes, hyperrectangles, hyperspheres) for approximating the
+local densities ``\\hat{f}(\\bf{x}_i)``, the `Lord` estimator uses hyperellopsoid volume
+elements. These hyperellipsoids are, for each query point `xᵢ`, estimated using singular
+value decomposition (SVD) on the `k`-th nearest neighbors of `xᵢ`. Thus, the hyperellipsoids
+stretch/compress in response to the local geometry around each sample point. This
+makes `Lord` a well-suited entropy estimator for a wide range of systems.
```

Comment:
Reading this, it seems to me that Lord is the most accurate estimator of the "true" underlying density (and hence diff entropy) of all the nearest neighbor based ones. Should we state this somewhere? E.g. in the table with a parenthesis (best)?

Reply:
It is actually not the most accurate estimator of differential entropy, because it doesn't apply any bias correction (which they explicitly state in the paper). In our estimator comparison in the documentation, you can actually see that `Lord` is negatively biased  (for the particular example we use) compared to many of the other estimators. 

However, the `Lord` estimator specifically appeared in the context of *mutual information* estimation. Judging from my simulations, in the mutual info context, for strongly correlated data it greatly outperforms methods such as KSG1/KSG2 dedicated diffmutualinfo estimators, and naive diffentropy estimators. 

I guess there is some cancellation effect going on, or something completely different. But they don't really analyse the theoretical properties of the estimator in that great detail, so I actually don't know. 

Take-away: we shouldn't claim that this estimator is "the best", just that is data-adaptive in a different manner than the estimators which assume constant density within some regular volume element like a square or circle.